### PR TITLE
Null safety migration

### PR DESCRIPTION
**Pull request by remonh87**
@werediver  the migration is not yet releasable but I want to merge this huge pr already in.

We are at the following state: 

- I did run the dart migration tool and fixed all issues the tool couldn't resolve
- I did the same for the example app
- All tests are green and no analysis issues
- The example app works

Next steps:

- We have to many nullable types in our api so I want to make sure I can get rid of them and make meaningful defaults.
-  I need to test more for the read write and subscribe so we need to extend our example app.

### CODE REVIEW

**Comment by remonh87** [comment]
@werediver  if you would like to review something I would suggest to review the last 5 (maybe 6 commits)

File: lib/src/model/connection_state_update.dart [hunk 1]
```diff
@@ -1,14 +1,11 @@
 import 'package:flutter_reactive_ble/src/model/generic_failure.dart';
-import 'package:functional_data/functional_data.dart';
 import 'package:meta/meta.dart';
 
-part 'connection_state_update.g.dart';
 //ignore_for_file: annotate_overrides
 
 ///Status update for a specific BLE device.
 @immutable
-@FunctionalData()
```
Comment (werediver) [inline lib/src/model/connection_state_update.dart:10]: Are you getting rid of functional_data in this p/r?

Comment (remonh87) [reply lib/src/model/connection_state_update.dart]: not anymore it was a temporary workaround but we are now using latest functional data which contains the fix.

**Review by werediver — APPROVED** [review]

**Comment by remonh87** [comment]
We need to wait for either: https://github.com/dart-lang/mockito/issues/312 or a new version of [mocktail](https://github.com/felangel/mocktail/issues)  before making the build green

**Comment by remonh87** [comment]
@werediver  the Dart team fixed the issue on mockito so we are now good to go!!

**Comment by werediver** [comment]
Good news! LGTM :shipit:

**Pull request merged**